### PR TITLE
Feat: add component create and edit

### PR DESCRIPTION
**Pull request by wangbow**
Signed-off-by: wb-wb665667 <wb-wb665667@alibaba-inc.com>

### Description of your changes

Fixes applicaiton Detail add component create and edit update

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [x] Run `yarn lint` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

Edit opetation:
image
Component list show:
image
Create Component:
image
image

Edit Component:
image

### CODE REVIEW

File: src/interface/application.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ export interface Trait {
   type: string;
   createTime?: string;
   updateTime?: string;
+  opetationAdd?: boolean;
```
Comment (barnettZQG) [inline src/interface/application.ts:183]: Why add this param?

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+  editComponent: any;
+  isEditComponent: boolean;
+  temporaryTraitList: Trait[];
+  componentDefinitions: [];
+  onAddTrait: () => void;
+  changeTraitStats: (is: boolean, params: any) => void;
+  onDeleteTrait: (type: string) => void;
+  onComponentOK: () => void;
+  onComponentClose: () => void;
+  dispatch?: ({}) => {};
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:36]: unused?

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+  t: (key: string) => {};
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:37]: There is no need to pass this function.  `i18n.t("xxx")`

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+import { Link } from 'dva/router';
+import locale from '../../../../utils/locale';
+import { If } from 'tsx-control-statements/components';
+import _ from 'lodash';
+import './index.less';
+
+type Props = {
+  appName?: string;
+  componentName?: string;
+  editComponent: any;
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:27]: Specify the correct type.

File: src/pages/ApplicationConfig/components/TraitDialog/index.tsx [hunk 5]
```diff
@@ -1,23 +1,27 @@
...
+  isEditComponent: boolean;
   isEditTrait: boolean;
   visible: boolean;
   traitItem?: Trait;
   traitDefinitions?: [];
   appName?: string;
   componentName?: string;
+  temporaryTraitList: Trait[];
+  createTemporaryTraitList: (trait: Trait) => void;
+  upDateTemporaryTraitList: (trait: Trait) => void;
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/TraitDialog/index.tsx:24]:
```suggestion
  updateTemporaryTraitList: (trait: Trait) => void;
```

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+        });
+        params.name = `${appName}-${name}`;
+        params.componentType = componentType;
+        params.traits = traitLists;
+        createApplicationComponent(params, { appName }).then((res) => {
+          if (res) {
+            Message.success({
+              duration: 4000,
+              title: 'Success',
+              content: 'create Component config success.',
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:113]:
```suggestion
              content: i18n.t('create Component config success'),
```

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+
+  extButtonList = () => {
+    const { onComponentClose, isEditComponent } = this.props;
+    return (
+      <div>
+        <Button type="secondary" onClick={onComponentClose} className="margin-right-10">
+          <Translation>Cancel</Translation>
+        </Button>
+        <If condition={!isEditComponent}>
+          <Button type="primary" onClick={this.onSubmitCreate}>
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:143]: Add loading

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+  getTraitList = () => {
+    const { isEditComponent, editComponent, temporaryTraitList } = this.props;
+    if (isEditComponent && editComponent) {
+      return [...(editComponent.traits || []), { opetationAdd: true }];
+    } else {
+      return [...temporaryTraitList, { opetationAdd: true }];
+    }
+  };
+
+  transComponentDefinitions() {
```
Comment (barnettZQG) [inline src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:189]: move to util package.

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx (hunk 3 above)
Comment (wangbow) [reply src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:42]: done. use i18.t('xxx')

File: src/pages/ApplicationConfig/components/TraitDialog/index.tsx (hunk 5 above)
Comment (wangbow) [reply src/pages/ApplicationConfig/components/TraitDialog/index.tsx]: done. change name.

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx (hunk 7 above)
Comment (wangbow) [reply src/pages/ApplicationConfig/components/ComponentDialog/index.tsx]: done

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx (hunk 8 above)
Comment (wangbow) [reply src/pages/ApplicationConfig/components/ComponentDialog/index.tsx]: done

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx (hunk 4 above)
Comment (barnettZQG) [reply src/pages/ApplicationConfig/components/ComponentDialog/index.tsx]: no change?

File: src/pages/ApplicationConfig/components/ComponentDialog/index.tsx (hunk 3 above)
Comment (barnettZQG) [reply src/pages/ApplicationConfig/components/ComponentDialog/index.tsx:42]: please remove the unuseful code.

**Review by barnettZQG — APPROVED** [review]
lgtm

**Pull request merged**